### PR TITLE
Adding nums package.

### PR DESCRIPTION
**Pull request by elibol**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml"
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged)
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details)
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

Hi,

Some questions:
1. The package requires Python >=3.6,<3.9. How do I enforce this?
2. I would like `mkl` to be a dependency for the package, which is not available in PyPi. Have I specified the dependencies properly?
3. In the future, what is the process for releasing new versions of my package?

### CODE REVIEW

**Comment by elibol** [comment]
@conda-forge-admin, please ping team conda-forge/help-python

**Comment by elibol** [comment]
NumS is closely related to Modin in terms of implementation and dependencies; I'll update the NumS yaml file to more closely resemble what's being done for Modin's conda forge package.

https://github.com/conda-forge/modin-feedstock/blob/master/recipe/meta.yaml

File: recipes/nums/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+  number: 0
+  noarch: python
+  entry_points:
+    - nums-coverage=nums.core.cmds.api_coverage:main
+  script: {{ PYTHON }} -m pip install . -vv
+
+requirements:
+  host:
+    - pip
+    - python >=3.6,<3.9
```
Comment (chrisburr) [inline recipes/nums/meta.yaml:23]: Does this actively not work on Python 3.9, or is it just dependencies that are missing? If it's just dependencies then it should be marked as below so it starts working once the dependencies are updated:
```suggestion
    - python >=3.6
```
(make sure to also update Python in `run` if you apply this)

File: recipes/nums/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+  run:
+    - mkl
+    - boto3
+    - scipy
+    - numpy >1.18.0,<=1.20.0
```
Comment (chrisburr) [inline recipes/nums/meta.yaml:28]: Does this really support 1.18.1 and 1.20.0 but not 1.18.0 and 1.20.1+? I guess this should actually be:
```suggestion
    - numpy >=1.18.0
```

File: recipes/nums/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+    - python >=3.6,<3.9
+    - ray >=1.0.0,<1.1.0
```
Comment (chrisburr) [inline recipes/nums/meta.yaml:30]: I guess this needs to be one of these: https://github.com/conda-forge/ray-packages-feedstock#current-release-info

File: recipes/nums/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+    - pytest
+    - moto
+    - tqdm
+  commands:
+    - pytest --pylint
+  requires:
+    - pytest==6.1.1
+    - pytest-pylint==0.17.0
+    - moto==1.3.16
+    - tqdm
```
Comment (chrisburr) [inline recipes/nums/meta.yaml:43]: I guess conda-forge doesn't care if the linting fails as that would be an upstream issue with the release?
```suggestion
  commands:
    - pytest
  requires:
    - pytest >=6.1.1
    - moto >=1.3.16
    - tqdm
```

File: recipes/nums/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+
+source:
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/nums-{{ version }}.tar.gz
+  sha256: 08ffe41a48615336399a92880304baab7c12c7761dc85749f05ef094d618b81a
+
+build:
+  number: 0
+  noarch: python
+  entry_points:
+    - nums-coverage=nums.core.cmds.api_coverage:main
```
Comment (chrisburr) [inline recipes/nums/meta.yaml:17]: Can you add a test for this command?

File: recipes/nums/meta.yaml [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+    - scipy
+    - numpy >1.18.0,<=1.20.0
+    - python >=3.6,<3.9
+    - ray >=1.0.0,<1.1.0
+
+test:
+  imports:
+    - pytest
+    - moto
+    - tqdm
```
Comment (chrisburr) [inline recipes/nums/meta.yaml:36]: None of these imports have anything to do with `nums` itself, how about:
```suggestion
    - nums
```

**Pull request closed without merging**